Backend/controllers/product.js

```diff
@@ -106,8 +98,61 @@ async function getProductsByUser(req, res) {
     }
 }
 
+async function deleteProduct(req, res) {
+    try {
+        const id = req.params.id;
+        const product = await products.findByIdAndDelete(id);
```

Comment:
We always avoid hard deletes in the application. For deleting, I suggest adding an additional field called "isActive".